src/ResourceManagement/Network/LoadBalancingRuleImpl.cs

```diff
@@ -339,5 +347,17 @@ internal LoadBalancingRuleImpl FromNewPublicIPAddress(string leafDnsLabel)
             Parent.WithNewPublicIPAddress(leafDnsLabel, frontendName);
             return FromFrontend(frontendName);
         }
+
+        internal LoadBalancingRuleImpl WithDisableOutboundSnat()
+        {
```

Comment:
If this two methods duplicated?

Reply:
The method without parameter is used in definition stage, when user not calling it, it is just as default true. The method with boolean parameter is used in update stage, user can change true to false, or false to true.